cmd/cluster/openstack/destroy.go

```diff
@@ -0,0 +1,67 @@
+package openstack
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/openshift/hypershift/cmd/cluster/core"
+	"github.com/openshift/hypershift/cmd/log"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/errors"
+)
+
+func NewDestroyCommand(opts *core.DestroyOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "openstack",
+		Short:        "Destroys a HostedCluster and its associated infrastructure on OpenStack",
+		SilenceUsage: true,
+	}
+
+	logger := log.Log
+	cmd.Run = func(cmd *cobra.Command, args []string) {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs, syscall.SIGINT)
+		go func() {
+			<-sigs
+			cancel()
+		}()
+
+		if err := DestroyCluster(ctx, opts); err != nil {
+			logger.Error(err, "Failed to destroy cluster")
+			os.Exit(1)
+		}
+	}
+
+	return cmd
+}
+func DestroyCluster(ctx context.Context, o *core.DestroyOptions) error {
+	hostedCluster, err := core.GetCluster(ctx, o)
+	if err != nil {
+		return err
+	}
+
+	if hostedCluster != nil {
+		o.InfraID = hostedCluster.Spec.InfraID
+	}
+
+	var inputErrors []error
+	if o.InfraID == "" {
+		inputErrors = append(inputErrors, fmt.Errorf("infrastructure ID is required"))
+	}
+	if err := errors.NewAggregate(inputErrors); err != nil {
+		return fmt.Errorf("required inputs are missing: %w", err)
+	}
+
+	return core.DestroyCluster(ctx, hostedCluster, o, destroyPlatformSpecifics)
+}
+
+func destroyPlatformSpecifics(ctx context.Context, o *core.DestroyOptions) error {
```

Comment:
Isn't there anything specific to delete from OpenStack? or once the HostedCluster is deleted, the openstack cluster also is and consequently OSP resources are wiped out?

Reply:
yeah everything is cleaned out, I checked